### PR TITLE
[SYCL] Add ext_oneapi_unified_runtime SYCL backend 

### PR DESCRIPTION
**Pull request by smaslov-intel**
This PR adds new backend to SYCL: ext_oneapi_unified_runtime
E2E test: https://github.com/intel/llvm-test-suite/pull/1450

NOTE: only review changes since https://github.com/intel/llvm/commit/fac63e9538995f497de42dfdf7a8f4785e3e15d5

The backend is being implemented by the plugin in https://github.com/intel/llvm/issues/7521
Plugin is still gated by https://github.com/intel/llvm/pull/7730

### CODE REVIEW

**Comment by smaslov-intel** [comment]
@againull : @sergey-semenov : please review https://github.com/intel/llvm/pull/7765/commits/fac63e9538995f497de42dfdf7a8f4785e3e15d5

File: sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero.cpp [hunk 1]
```diff
@@ -390,3 +390,57 @@ ZER_APIEXPORT zer_result_t ZER_APICALL zerPlatformGet(
...
+    zer_platform_handle_t Platform, ///< [in] handle of the platform
+    zer_platform_info_t ParamName,  ///< [in] type of the info to retrieve
+    size_t *pSize, ///< [in,out] pointer to the number of bytes needed to return
+                   ///< info queried. the call shall update it with the real
+                   ///< number of bytes needed to return the info
+    void *ParamValue ///< [out][optional] array of bytes holding the info.
+                     ///< if *pSize is not equal to the real number of bytes
+                     ///< needed to return the info then the
+                     ///< ::ZER_RESULT_ERROR_INVALID_SIZE error is returned and
+                     ///< pPlatformInfo is not used.
```
Comment (againull) [inline sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero.cpp:404]: I am not against it but is this some known format of commenting API? I think we should use capital case at the beginning of the sentences.

File: sycl/plugins/unified_runtime/pi_unified_runtime.cpp [hunk 2]
```diff
@@ -6,12 +6,79 @@
...
+                                          size_t *ParamValueSizeRet) {
+  return pi2ur::piPlatformGetInfo(Platform, ParamName, ParamValueSize,
+                                  ParamValue, ParamValueSizeRet);
+}
+
+__SYCL_EXPORT pi_result piDevicesGet(pi_platform Platform,
+                                     pi_device_type DeviceType,
+                                     pi_uint32 NumEntries, pi_device *Devices,
+                                     pi_uint32 *NumDevices) {
+  // Report no devices, stab to have a minimal SYCL test running
```
Comment (againull) [inline sycl/plugins/unified_runtime/pi_unified_runtime.cpp:40]: stab->stub

**Comment by againull** [comment]
Just minor comments, other than that https://github.com/intel/llvm/commit/fac63e9538995f497de42dfdf7a8f4785e3e15d5 looks good to me.

File: sycl/source/detail/platform_impl.cpp [hunk 3]
```diff
@@ -126,8 +126,8 @@ std::vector<platform> platform_impl::get_platforms() {
           // insert PiPlatform into the Plugin
           Plugin.getPlatformId(PiPlatform);
         }
-        if (!Platform.get_devices(info::device_type::all).empty())
-          Platforms.push_back(Platform);
+        // if (!Platform.get_devices(info::device_type::all).empty())
+        Platforms.push_back(Platform);
```
[the remark this answers is not in the document]
Comment (smaslov-intel) [reply sycl/source/detail/platform_impl.cpp:130]: That's right, no devices. And I couldn't find in SYCL spec that platforms without devices should not be exposed.

File: sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero.cpp (hunk 1 above)
Comment (smaslov-intel) [reply sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero.cpp:404]: This is the format used in parent https://github.com/oneapi-src/unified-runtime/blob/cdb62e54bba8f9e56d3438e0d3d91e3037862f55/include/ur_api.h#L3709

I'll let the new owners choose/reformat as needed
The Unified Runtime is to become an external component for SYCL

File: sycl/plugins/unified_runtime/pi_unified_runtime.cpp (hunk 2 above)
Comment (smaslov-intel) [reply sycl/plugins/unified_runtime/pi_unified_runtime.cpp:40]: OK

File: sycl/plugins/unified_runtime/pi_unified_runtime.cpp [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
```
Comment (smaslov-intel) [inline sycl/plugins/unified_runtime/pi_unified_runtime.cpp:40]:
```suggestion
  // Report no devices, stub to have a minimal SYCL test running

```

**Comment by smaslov-intel** [comment]
moved to https://github.com/intel/llvm/pull/7816

**Pull request closed without merging**